### PR TITLE
Refactor notes logic out of the persistence method

### PR DESCRIPTION
**Pull request by codyseibert**

### CODE REVIEW

File: shared/src/business/useCases/caseNote/getUserCaseNoteForCasesInteractor.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+} = require('../../../authorization/authorizationClientService');
+const { UnauthorizedError } = require('../../../errors/errors');
+const { UserCaseNote } = require('../../entities/notes/UserCaseNote');
+
+/**
+ * getUserCaseNoteForCasesInteractor
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @param {string} providers.caseId the id of the case to get notes for
```
Comment (kkoskelin) [inline shared/src/business/useCases/caseNote/getUserCaseNoteForCasesInteractor.js]: *`caseIds`

File: shared/src/proxies/caseNote/getUserCaseNoteForCasesProxy.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+const { get } = require('../requests');
+
+/**
+ * getUserCaseNoteForCasesInteractor
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext the application context
+ * @param {string} providers.caseId the case id to get notes for the logged in user
```
Comment (kkoskelin) [inline shared/src/proxies/caseNote/getUserCaseNoteForCasesProxy.js]: *`caseIds`

File: web-api/src/caseNote/getUserCaseNoteForCasesLambda.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const { genericHandler } = require('../genericHandler');
+
+/**
+ * used for fetching a judge's case note
```
Comment (kkoskelin) [inline web-api/src/caseNote/getUserCaseNoteForCasesLambda.js:4]: it's for a user, not a judge, yes?

File: web-client/src/presenter/actions/TrialSession/getUserCaseNoteForCasesAction.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Fetches the cases notes
+ *
+ * @param {object} providers the providers object
+ * @param {object} providers.applicationContext needed for getting the getCase use case
+ * @param {object} providers.props the cerebral props object containing the props.trialSessionId
+ * @returns {object} contains the eligible cases for a trial sessions
```
Comment (kkoskelin) [inline web-client/src/presenter/actions/TrialSession/getUserCaseNoteForCasesAction.js]: jsdoc update

**Review by kkoskelin — APPROVED** [review]
Some jsdoc updates required.  Also I'm wondering if all the new files have coverage yet?

File: web-api/src/caseNote/getUserCaseNoteForCasesLambda.js (hunk 3 above)
Comment (codyseibert) [reply web-api/src/caseNote/getUserCaseNoteForCasesLambda.js:4]: internally "user-notes" are judge's notes; it's confusing.  We probably need to revisit the naming of this stuff at some point

Comment (rachaelparris) [reply web-api/src/caseNote/getUserCaseNoteForCasesLambda.js:4]: User notes can be judge's notes or trial clerk's notes - hence "user notes". They used to be called judge's notes and that wasn't truthful anymore when we added trial clerk notes.

**Review by kkoskelin — APPROVED** [review]

**Pull request merged**